builtin/providers/nsone/provider.go

```diff
@@ -32,9 +35,10 @@ func Provider() terraform.ResourceProvider {
 }
 
 func nsoneConfigure(d *schema.ResourceData) (interface{}, error) {
-	n := nsone.New(d.Get("apikey").(string))
-	n.Debug()
-	n.RateLimitStrategySleep()
+	httpClient := &http.Client{}
+	n := nsone.NewClient(httpClient, nsone.SetAPIKey(d.Get("apikey").(string)))
+	// FIXME: n.Debug()
```

Comment:
i wasnt sure what to do here either. `v1` prints requests and responses when debug is enabled.

we can add a logger that logs every request with the following if that is what users would want.
```
	// Adds logging to each http request.
	doer := api.Decorate(httpClient, api.Logging(log.New(os.Stdout, "", log.LstdFlags)))
	client := api.NewClient(doer, api.SetAPIKey(k))
```